packages/url-launcher/android/src/main/java/io/flutter/plugins/url_launcher/UrlLauncherPlugin.java

```diff
@@ -13,34 +18,51 @@
  * UrlLauncherPlugin
  */
 public class UrlLauncherPlugin implements MethodCallHandler {
-  private FlutterActivity activity;
-
-  public static UrlLauncherPlugin register(FlutterActivity activity) {
-    return new UrlLauncherPlugin(activity);
-  }
-
-  private UrlLauncherPlugin(FlutterActivity activity) {
-    this.activity = activity;
-    new MethodChannel(
-            activity.getFlutterView(), "plugins.flutter.io/URLLauncher").setMethodCallHandler(this);
-  }
-
-  @Override
-  public void onMethodCall(MethodCall call, Result result) {
-    if (call.method.equals("UrlLauncher.launch")) {
-      launchURL((String) call.arguments);
-      result.success(null);
-    } else {
-      result.notImplemented();
+    private FlutterActivity activity;
+
+    public static UrlLauncherPlugin register(FlutterActivity activity) {
+        return new UrlLauncherPlugin(activity);
     }
-  }
-  private void launchURL(String url) {
-    try {
-      Intent launchIntent = new Intent(Intent.ACTION_VIEW);
-      launchIntent.setData(Uri.parse(url));
-      activity.startActivity(launchIntent);
-    } catch (java.lang.Exception exception) {
-      // Ignore parsing or ActivityNotFound errors
+
+    private UrlLauncherPlugin(FlutterActivity activity) {
+        this.activity = activity;
+        new MethodChannel(
+                activity.getFlutterView(), "plugins.flutter.io/URLLauncher").setMethodCallHandler(this);
+    }
+
+    @Override
+    public void onMethodCall(MethodCall call, Result result) {
+        String url = call.arguments();
+        if (call.method.equals("UrlLauncher.canLaunch")) {
+            canLaunch(url, result);
+        } else if (call.method.equals("UrlLauncher.launch")) {
+            launchURL(url, result);
+        } else {
+            result.notImplemented();
+        }
+    }
+
+    private void launchURL(String url, Result result) {
+        try {
+            Intent launchIntent = new Intent(Intent.ACTION_VIEW);
+            launchIntent.setData(Uri.parse(url));
+            activity.startActivity(launchIntent);
+            result.success(null);
+        } catch (java.lang.Exception exception) {
+            result.error("ERROR", exception.getMessage(), null);
+        }
+    }
+
+    private void canLaunch(String url, Result result) {
+        Intent launchIntent = new Intent(Intent.ACTION_VIEW);
+        launchIntent.setData(Uri.parse(url));
+        ComponentName componentName = launchIntent.resolveActivity(activity.getPackageManager());
+        if (componentName == null ||
+                "{com.android.fallback/com.android.fallback.Fallback}".
+                        equals(componentName.toShortString())) {
+            result.success(false);
+        } else {
+            result.success(true);
+        }
```

Comment:
Nit: `if (bExp) { xxx(true); } else { xxx(false); }` should often be replaced by `xxx(bExp);`

Here, `bExp` is quite long and so it may be more readable to do

```
boolean answer = bExp;
xxx(answer);
```